src/ORM/UnionList.php

```diff
@@ -0,0 +1,226 @@
+{
+    /**
+     * An array of SS_List instances
+     *
+     * @var array
+     */
+    protected $lists = array();
+
+    /**
+     * Create a new UnionList.
+     *
+     * @param array $lists - Lists to hold
+     */
+    public function __construct(array $lists)
+    {
+        $this->lists = array_values($lists);
+        parent::__construct();
+    }
+
+    /**
+     * Return the total number of items in each SS_List.
+     *
+     * @return int
+     */
+    public function count()
+    {
+        $count = 0;
+        foreach ($this->lists as $list) {
+            $count += $list->count();
+        }
+        return $count;
+    }
+
+     /**
+     * Returns true if one of the SS_List's have an item.
+     *
+     * @return bool
+     */
+    public function exists()
+    {
+        return $this->count() > 0;
+    }
+
+    /**
+     * Iterate over each SS_List, one after the other.
+     *
+     * @return ArrayIterator
+     */
+    public function getIterator()
+    {
+        foreach ($this->lists as $list) {
+            foreach ($list as $record) {
+                yield $record;
+            }
+        }
+    }
+
+    /**
+     * Get array of each record in each SS_List.
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        $result = array();
+        foreach ($this->lists as $list) {
+            foreach ($list as $record) {
+                $result[] = $record;
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Walks the list using the specified callback
+     *
+     * @param callable $callback
+     * @return $this
+     */
+    public function each($callback)
+    {
+        foreach ($this as $item) {
+            $callback($item);
+        }
+        return $this;
+    }
+
+    public function debug()
+    {
+        $val = "<h2>" . $this->class . "</h2><ul>";
+        foreach ($this->toNestedArray() as $item) {
+            $val .= "<li style=\"list-style-type: disc; margin-left: 20px\">" . Debug::text($item) . "</li>";
+        }
+        $val .= "</ul>";
+        return $val;
+    }
+
+    /**
+     * Return this list as an array and every object it as an sub array as well
+     *
+     * @return array
+     */
+    public function toNestedArray()
+    {
+        $result = array();
+        foreach ($this->lists as $list) {
+            foreach ($list as $item) {
+                if (is_object($item)) {
+                    if (method_exists($item, 'toMap')) {
+                        $result[] = $item->toMap();
+                    } else {
+                        $result[] = (array) $item;
+                    }
+                } else {
+                    $result[] = $item;
+                }
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Get first item in the first SS_List.
+     *
+     * @return array
+     */
+    public function first()
+    {
+        $list = resets($this->lists);
```

Comment:
I know.

Just an FYI, I'm generally working on unit tests before I go out, and the easiest way to ensure I can access it from a different house/computer is just pushing it up. Didn't need feedback here :)